### DISCUSSION TITLE
enforcing character limit while typing or copying and pasting text

### DISCUSSION THREAD

**Discussion by yuvaraj-thiagarajan** [remirror/remirror#903] [Q&A]
Is there a way to enforce character limit while typing or pasting content from somewhere else? I am looking for an extension that trims longer text to a specified maximum length while copying and pasting content in addition to not allowing the users to type beyond a set character limit.

**Comment by Shubhdeep12** [comment]
> Is there a way to enforce character limit while typing or pasting content from somewhere else? I am looking for an extension that trims longer text to a specified maximum length while copying and pasting content in addition to not allowing the users to type beyond a set character limit.

Hi, did you get any extension for this?

**Comment by KSABCarl** [comment]
I made a  character count extension with character limit that works alright at least for simpler documents. Have not tested it extensively but may be something to start out with?
```typescript
import {
  extension,
  helper,
  PlainExtension,
  EditorState,
  CreateExtensionPlugin,
  Helper,
} from '@remirror/core';

export interface CharacterCountOptions {
  limit?: number;
}

@extension<CharacterCountOptions>({
  defaultOptions: {
    limit: 500,
  },
})
export class CharacterCountExtension extends PlainExtension<CharacterCountOptions> {
  get name() {
    return 'characterCount' as const;
  }

  createPlugin(): CreateExtensionPlugin {
    const { limit } = this.options;
    return {
      appendTransaction: (_transactions, _oldState, newState) => {
        const length = newState.doc.content.size;
        if (limit && length > limit) {
          return newState.tr.insertText('', limit + 1, length);
        }
      },
    };
  }
  @helper()
  getCharacterCount(
    state: EditorState = this.store.getState()
  ): Helper<number> {
    return state.doc.content.size - 2;
  }
}

declare global {
  namespace Remirror {
    interface AllExtensions {
      characterCount: CreateExtensionPlugin;
    }
  }
}

```
